internal/controlplane/handlers_oauth.go

```diff
@@ -92,6 +95,13 @@ func (s *Server) GetAuthorizationURL(ctx context.Context,
 		owner = sql.NullString{Valid: true, String: *req.Owner}
 	}
 
+	var redirectUrl sql.NullString
+	if req.RedirectUrl == nil {
+		redirectUrl = sql.NullString{Valid: false}
+	} else {
+		redirectUrl = sql.NullString{Valid: true, String: *req.RedirectUrl}
```

Comment:
In this case the redirect URL will be something like `minder.com/onboarding/repo-registration` so that the frontend user can continue where they left off before the redirect to GitHub. It won't have any additional query params like `code` etc. Do you think it still needs to be encrypted?

Reply:
gotcha, it wasn't entirely clear to me what it would contain. If it's just the host and path it's fine.

Reply:
Should we add URL validation so we make sure such things don't get stored here then?

Reply:
I'd be more inclined to encrypt it just in case, rather than restrict what can be passed in. The frontend team hasn't determined what the structure will be, and I'd like to allow the flexibility of sending `minder.com/onborading?step=repo-registration` if that ends up being more convenient for them. What do you think?

Reply:
Can you give a little bit more information on what the proposed flow and usage for this is? Just wanna make sure we don't over-complicate things while keeping the security concerns in check.

Reply:
For sure, this is the intended flow:
- User wants to enrol a provider from the frontend Minder app
- FE queries BE to get the github authorization URL providing the URL where the Minder backend will send the user when enrolment is complete, for example `minder.com/onboarding/repo-registration`
- BE responds with the GitHub authorization URL and the frontend redirects the user there
- When GitHub authorization is complete, GitHub redirects to Minder BE with the oauth code and state param. Minder BE uses the code to generate an OAuth token, and stores the token. It uses the state to figure out 1) which project to associate the token with and 2) to get the redirect back to the frontend
- Minder BE redirects the user back to the frontend

The key difference between the web flow and the CLI flow is that the CLI flow shows the "Minder enrollment complete" page, while the web flow takes the user back to the frontend.

With the web flow and the CLI flow, none of the OAuth data (code or token) is passed to the frontend. It's all handled exclusively by the backend.

Reply:
The redirect URI is stored in the DB, which is what I suggested encrypting.

Reply:
Ah - I'd missed that because it was off the diff. Carry on!